### DISCUSSION TITLE
[Mobile] [IOS] Quill Cursor took the initial position & style when tab anywhere after the first time and start typing.

### DISCUSSION THREAD

**Discussion by AkshayJainX** [singerdmx/flutter-quill#554] [Mobile] [answered]
My issue is about [Mobile] with IOS. I'm facing an issue with IOS 15. **Also, the issue started when quill upgraded to the Flutter 2.8 version.**

I have tried running the `example` directory successfully before creating an issue here. 

When I placed the cursor for the first time, it works. But when I placed it for the next time and start typing. It starts from the last cursor position. I've added the video of the error.

Do let me know if you've any queries. 

Flutter & Dart Version - 
Flutter 2.8.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 77d935af4d (9 days ago) • 2021-12-16 08:37:33 -0800
Engine • revision 890a5fca2e
Tools • Dart 2.15.1

https://user-images.githubusercontent.com/36562389/147393980-452cfae4-40ae-47ff-bb12-79e0aa635309.mov

**Comment by AkshayJainX** [comment]
Hi @singerdmx, 

Happy New Year 2022!

This issue is not resolved in 3.0.4. I'm still facing it in Flutter 2.8.

If you need any further info. Do let me know.

**Reply by AkshayJainX** [reply]
Hi @singerdmx, @cbracken, @rish07, @yzxben, @jonmountjoy,

This issue still persists in IOS 15 for flutter_quill: ^3.2.0 version.

**Reply by AkshayJainX** [reply]
Also, I found that issue is with the external keyword. So what I found is that when I type through the iPhone soft keyword it works but when I use the mac keyword to type in the IOS simulator the issue exists.

**Comment by mark8044** [comment]
I am also running into this problem more and more, mostly on the iOS side and iOS simulator.  I'm not sure if its a problem with the way I've implemented flutter_quill or a problem with the package itself?

**Comment by MagTuxGit** [comment]
@singerdmx 
Here both `ignoreFocus` and `_keyboardVisible` are false so remote selection doesn't change.
https://github.com/singerdmx/flutter-quill/blob/4f36393aeac311cd4e10c40819c136e7734c9738/lib/src/widgets/raw_editor.dart#L674-L684

**Reply by singerdmx** [reply]
OK, could you make a fix please?

**Reply by MagTuxGit** [reply]
I need some time to figure out what's going on here. 
E.g. why can't we just call `updateRemoteValueIfNeeded();` every time something changes

**Reply by singerdmx** [reply]
Sorry I don't remember

**Reply by MagTuxGit** [reply]
I had an idea to check for iOS Simulator, but it's an async call and it's not good to use it here.

**Reply by MagTuxGit** [reply]
Ok, I'll make a little research and make a fix

**Reply by MagTuxGit** [reply]
iOS Simulator check is also not good because the issue can appear on iPad with external keyboard or on a desktop.

**Reply by MagTuxGit** [reply]
I don't have an M1 Macbook to check it.

**Reply by mark8044** [reply]
I think it shows up well in iOS simulator because you are inputing from your keyboard.  If you use the soft keyboard on the simulator directly then its not a problem.  In fact, when I type quickly into the simulator from my physical keyboard the problem frequently shows up, and to correct it I press command-K to bring up the soft keyboard, press a single button on it and the problem goes away for a few seconds until it comes back again after typing a few words

**Comment by singerdmx** [comment, marked as answer]
fixed by @MagTuxGit

**Reply by mark8044** [reply]
I'm on 3.9.10 and still seeing this problem in the iOS simulator

**Reply by singerdmx** [reply]
> I'm on 3.9.10 and still seeing this problem in the iOS simulator

please debug it

**Reply by MagTuxGit** [reply]
> I'm on 3.9.10 and still seeing this problem in the iOS simulator

Can you specify what exactly problem are you still seeing?

**Reply by MagTuxGit** [reply]
https://user-images.githubusercontent.com/5296407/152956837-600fc8ec-7028-4024-bda4-11b148a1c737.mov

**Reply by mark8044** [reply]
The problem I'm seeing is probably different.  Whats happening is that I'm inserting text into the quill controller as follows:
```
_controller.document.insert(_curStartPos, widget.quotetext)+'\n');
_controller.document.format(_curStartPos,widget.quotetext.length, Attribute.blockQuote);
_controller.moveCursorToEnd();
```

So that when you open the editor you see a block quote of text and your cursor is sitting on a new fresh line.

As soon as you start typing, the cursor immediately jumps to position 1 (the first spot within the block quote text that was prepopulated).


As it sounds like is a separate issue, when you are typing a lot of text quickly into the simulator, the cursor will seemingly randomly jump to different places, or just freeze.  Opening the soft keyboard would allow you to unfreeze, reposition and properly type for a short while.

I thought these were all linked to the same underlying problem, but maybe not
